src/decl/decl-ok/impl-item.rkt

```diff
@@ -78,12 +79,37 @@
    (where/error (Goal_bty ...) (where-clauses->goals CrateDecls WhereClauses_bty))
 
    ; goals (a) and (b) -- the type `Ty` meets its bounds and is well-formed,
-   ; assuming the where-clauses on the assoc type impl item hold
+   ; assuming the where-clauses on the assoc type impl item hold.
+   ;
+   ; IMPORTANT: We use `well-formed-goal-for-ty` to create the goal, rather than
+   ; writing `(well-formed (type Ty))`. This "breaks the loop" to prevent us from
+   ; leveraging this very impl to prove WF. Imagine this program:
+   ;
+   ; ```
+   ; impl Iterator for SomeType { type Item = SomeOtherType<String>; }
+   ; struct SomeOtherType<T: Copy> { t: T }
+   ; ```
+   ;
+   ; This other type `SomeOtherType<String>` is not well-formed, so the impl should be invalid.
+   ; But if we just tried to prove `(well-formed (type SomeOtherType<String>))`, the solver would be
+   ; within its rights to combine the facts that ...
```

Comment:
```suggestion
   ; However, if we tried to prove `(well-formed (type SomeOtherType<String>))` directly, the solver would be
   ; within its rights to combine the facts that ...
```

Maybe something like "the solver could (erroneously) combine the facts" or "the solver might (erroneously) combine the facts"? Would the solver definitely make this mistake given this example, or is it a more general concern that depends on other details as well?
```suggestion
   ; But if we just tried to prove `(well-formed (type SomeOtherType<String>))`, the solver would be
   ; within its rights to combine the facts that
```